component/core/src/test/java/org/exoplatform/social/core/listeners/FileAttachmentListenerTest.java

```diff
@@ -0,0 +1,167 @@
+    fileAttachmentListener = new FileAttachmentListener(fileService);
+    if (metadataService.getMetadataTypeByName(METADATA_TYPE_NAME) == null) {
+      MetadataTypePlugin userMetadataTypePlugin = new MetadataTypePlugin(newParam(2659, METADATA_TYPE_NAME)) {
+        @Override
+        public boolean isAllowMultipleItemsPerObject() {
+          return true;
+        }
+
+        @Override
+        public boolean isShareable() {
+          return true;
+        }
+      };
+      metadataService.addMetadataTypePlugin(userMetadataTypePlugin);
+    }
+
+    johnIdentity = identityManager.getOrCreateUserIdentity("john");
+    tearDownActivityList = new ArrayList<>();
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    end();
+    begin();
+    identityManager.deleteIdentity(johnIdentity);
+    metadataDAO.deleteAll();
+
+    for (ExoSocialActivity activity : tearDownActivityList) {
+      activityManager.deleteActivity(activity.getId());
+    }
+    super.tearDown();
+  }
+
+  public void testDeleteActivity() throws Exception {
+    String activityTitle = "activity title";
+    String userId = johnIdentity.getId();
+    ExoSocialActivity activity = new ExoSocialActivityImpl();
+    activity.setTitle(activityTitle);
+    activity.setUserId(userId);
+    activityManager.saveActivityNoReturn(johnIdentity, activity);
+
+    long creatorId = Long.parseLong(johnIdentity.getId());
+    MetadataKey metadataKey = new MetadataKey(METADATA_TYPE_NAME, String.valueOf(FILE_ID), AUDIENCE_ID);
+    MetadataObject metadataObject = new MetadataObject(ExoSocialActivityImpl.DEFAULT_ACTIVITY_METADATA_OBJECT_TYPE,
+                                                       activity.getId(),
+                                                       PARENT_OBJECT_ID);
+    MetadataItem metadataItem = metadataService.createMetadataItem(metadataObject, metadataKey, creatorId);
+    assertNotNull(metadataItem);
+    assertTrue(metadataItem.getId() > 0);
+
+    List<MetadataItem> metadataItems = metadataService.getMetadataItemsByObject(metadataObject);
+    assertNotNull(metadataItem);
+    assertEquals(1, metadataItems.size());
+
+    FileItem fileItem = new FileItem(FILE_ID,
+                                     "test",
+                                     "image/png",
+                                     "attachment",
+                                     "test".getBytes().length,
+                                     new Date(),
+                                     "user",
+                                     false,
+                                     new ByteArrayInputStream("test".getBytes()));
+
+    fileItem = fileService.writeFile(fileItem);
+
+    activityManager.deleteActivity(activity);
+    restartTransaction();
+
+    Event<Long, List<MetadataItem>> deleteAttachments = new Event<>("social.metadataItem.deleted", 0l, metadataItems);
+    fileAttachmentListener.onEvent(deleteAttachments);
```

Comment:
The test is fake, in fact, this should be triggered automatically when the Activity is deleted.
```suggestion
```